src/recipes/tests/recipes.service.spec.ts

```diff
@@ -0,0 +1,95 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { RecipesService } from '../recipes.service';
+import { GetRecipesQueryParamsDto } from '../dto/get-recipe-query-params.dto';
+import { ForbiddenException, NotFoundException } from '@nestjs/common';
+import { mockedGetRecipesResponse } from './mocks/get-recipes-response.mock';
+import { mockedGetRecipeResponse } from './mocks/get-recipe-response.mock';
+import { GetRecipeResponse } from '../dto/get-recipe-response';
+import { mockedGetRecipesQueryParams } from './mocks/recipe-query-params.mock';
+
+describe('RecipesService', () => {
+  let service: RecipesService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        {
+          provide: RecipesService,
+          useValue: {
+            getRecipes: jest.fn(),
+            getRecipeById: jest.fn(),
+          },
+        },
+      ],
+    }).compile();
+
+    service = module.get<RecipesService>(RecipesService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+  describe('getRecipes', () => {
+    it('should call getRecipes with the correct userId and queryParams and return an array of GetRecipeResponse', async () => {
+      const userId = '1';
+      const queryParams: GetRecipesQueryParamsDto = mockedGetRecipesQueryParams;
+
+      const result = mockedGetRecipesResponse;
+      jest.spyOn(service, 'getRecipes').mockResolvedValue(result);
+
+      expect(await service.getRecipes(userId, queryParams)).toEqual(result);
+      expect(service.getRecipes).toHaveBeenCalledWith(userId, queryParams);
+    });
+
+    it('should throw ForbiddenException if user is not authorized', async () => {
+      const userId = '1';
+      const queryParams: GetRecipesQueryParamsDto = mockedGetRecipesQueryParams;
+
+      jest.spyOn(service, 'getRecipes').mockImplementation(async () => {
+        throw new ForbiddenException('User is not authorized');
+      });
+
+      await expect(service.getRecipes(userId, queryParams)).rejects.toThrow(
+        ForbiddenException,
+      );
+    });
+  });
+
+  describe('getRecipeById', () => {
```

Comment:
Another hint - I recommend exporting each describe block (which in this case is a different service method which is very good 👍🏻 ) to a separate files. This way you will avoid test files that are 1000+ lines of code in case of bigger services.